### ISSUE TITLE
[BUG] Digital gauge current: value resets to zero after page finishes loading

### ISSUE THREAD

**Issue by ghost** [thingsboard/thingsboard#1492]
Hi there, 

I have an issue with the digital gauges. When page first loads the gauge loads with the current value appearing as 0. When the page is manually refreshed the value appears for a short period of time before resetting and staying at 0, once the page is finished loading.

Video of bug happening on a dashboard:

[https://youtu.be/W99Dxzxmn2A](url)

I'm on Chrome 72.0.3626.109

Possible duplicate of # 856 [https://github.com/thingsboard/thingsboard/issues/856](url) ??

**Comment by lepgithub** [comment]
@enviroeye 
Are you having the same issue with analog gauge widgets?

**Comment by enviroeye** [comment]
No it seems to only affect the digital gauges

**Comment by AlexanderJP91** [comment]
I also discovered this bug that was arising whenever I has entering fullscreen mode. I've checked the Widget Library and found out that this arises whenever an `onResize` event is triggered. I then checked out the widget source code, and the resize event invokes this call:
https://github.com/thingsboard/thingsboard/blob/ced63c4e70579f9567e05e1632322f226c7cce19/ui/src/app/widget/lib/canvas-digital-gauge.js#L226-L228
Where `this.gauge` is an instance of the `CanvasDigitalGauge` class:
https://github.com/thingsboard/thingsboard/blob/ced63c4e70579f9567e05e1632322f226c7cce19/ui/src/app/widget/lib/canvas-digital-gauge.js#L193
I then found out that the `update` method invoked by `this.gauge` is the following: https://github.com/thingsboard/thingsboard/blob/ab494596e7e859a73df58e1c9aca6ffa3da8ea92/ui/src/app/widget/lib/CanvasDigitalGauge.js#L148-L155
At this point I just suppose that the invocation of `this.initValueClone();` resets the value, but I'm not sure about this.
Anyhow There is no possible way to modify the `CanvasDigitalGauge` class on an already generated platform, I does require to modify the source code and regenerate the bundle.
I though find a work around for this problem. Just generate a new Widget starting from the digital Gauge base and change the `self.onResize` event from:
```
self.onResize = function() {
    self.ctx.gauge.resize();
}
```
To:
```
self.onResize = function() {
    //self.ctx.gauge.resize();
    delete self.ctx.gauge;
    self.onInit();
    self.onDataUpdated();
}
```
This will practically delete the current digital gauge and regenerate it and update it. You can see this effect when resizing (or going in fullscreen), the digital gauge will totally regenerate.

**Comment by Ultrazombie** [comment]
We inform you that your GitHub ticket has been inactive for some time, and we will close it. If you have any more concerns or encounter similar issues, please don't hesitate to contact us.
